main.py

```diff
@@ -67,41 +69,64 @@ async def marco(ctx):
     @bot.command()
     async def membercardsranked(ctx, *args):
         invalid_cmd_reply = 'Please provide a clan tag. E.g.\n`!membercardsranked 9GULPJ9L`'
+        card_type_arg = 'all'
         if len(args) == 0:
             await ctx.send(invalid_cmd_reply)
             return
+        elif len(args) == 2:
+            card_type_arg = args[1].rstrip('s')
+            valid_arguments = schemas.CARD_TYPE_ID_PREFIX.values()
+            if card_type_arg not in valid_arguments:
+                message = 'Second argument is not in scope, please provide the correct card filter: troops, spells, or buildings'
+                await ctx.send(embed=discord.Embed(
+                    description=message,
+                    colour=discord.Colour.red()
+                ))
+                return
         clan_tag = args[0]
 
+        await fetch_ranked_members(ctx, clan_tag, card_type_arg)
+
+    @ bot.command()
+    async def ausclan(ctx, *args):
+        card_type_arg = 'all'
+        if len(args) == 1:
+            card_type_arg = args[0].rstrip('s')
+            valid_arguments = schemas.CARD_TYPE_ID_PREFIX.values()
+            if card_type_arg not in valid_arguments:
+                message = 'First argument is not in scope, please provide the correct card filter: troops, spells, or buildings'
+                await ctx.send(embed=discord.Embed(
+                    description=message,
+                    colour=discord.Colour.red()
+                ))
+                return
```

Comment:
Weren't you going to make this into a function so that you don't have duplicate code in `membercardsranked` and `ausclan`? Can make it return an embed. And in the caller function, only send error message and early return if invalid argument

Reply:
done by creating a function: card_type_check